state/core/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityViews.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) 2018-2022 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.state;
+
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.Event;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.Specification;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static lombok.AccessLevel.PACKAGE;
+
+public final class EntityViews {
+
+  @NonNull
+  public static EntityView defaultEntityView(EventReceiver receiver) {
+    return new DefaultEntityView(receiver);
+  }
+
+  @NonNull
+  public static EntityView meteredEntityView(EventReceiver receiver, MeterRegistry meterRegistry) {
+    Map<Entity.Key<?>, StateValue> entities = new ConcurrentHashMap<>();
+    meterRegistry.gaugeMapSize("stream_registry_state.view.entities", Tags.empty(), entities);
+
+    DefaultEntityViewUpdater defaultEntityViewUpdater = new DefaultEntityViewUpdater(entities);
+    return new DefaultEntityView(receiver, entities, new MeteredEntityViewUpdater(defaultEntityViewUpdater, meterRegistry));
+  }
+
+  @RequiredArgsConstructor(access = PACKAGE)
+  static final class MeteredEntityViewUpdater implements EntityViewUpdater {
+    @NonNull
+    private final EntityViewUpdater delegate;
+    @NonNull
+    private final MeterRegistry meterRegistry;
+
+    @Override
+    public <K extends Entity.Key<S>, S extends Specification> Entity<K, S> update(Event<K, S> event) {
+      meterRegistry.counter("stream_registry_state.receiver.update", Tags.of("event", simpleName(event))).increment();
```

Comment:
Could we add more tags on these? Possible Key type and the specification type field. It's a shame that URNs are not part of the OOS project, they would ultimately be very useful here. I guess we could implement that in a more advanced metered updater in other projects.

Reply:
I've extended tags to include the event type and the key. I've not added more than that (key names etc) since it could greatly increase the cardinality, which in turn could increase costs for some observability platforms (e.g. DataDog is costs per unique metric name/tag value)